### PR TITLE
Creation of wrapper for merqury histogram plot

### PR DESCRIPTION
**Pull request by Delphine-L**
This wrapper is for the tool plot_spectra_cn.R included in merqury. The reason to isolate it is to allow the re-plotting of the spectra with different options than the ones by default in the merqury tool.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/merqury/merqury_plot_CN.xml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+<tool id="merquryplot" name="Merqury histogram plot" version="@TOOL_VERSION@+galaxy1" profile="20.01">
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:1]: Use the version suffix token?

File: tools/merqury/merqury_plot_CN.xml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+            -n $ymax
+        #end if
+        -t $type
+        #if $output_selector=='pdf'
+            -p
+        #end if 
+
+    ]]>    </command>
+    <inputs>
+        <param type="data" name="input1" format="tab" />
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:36]: Should the format be tabular?

File: tools/merqury/merqury_plot_CN.xml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        <param name="xdim" type="integer" label="Width of plot"  value="6" min="0" help="(-x)" />
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:37]: Use argument instead of listing them in the help

File: tools/merqury/merqury_plot_CN.xml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        <param name="ydim" type="integer" label="Height of plot"  value="5" min="0"  help="(-y)" />
+        <param name="xmax" type="float" label="Maximum limit for k-mer multiplicity (leave 0 for default)"  value="0" min="0" help="(-m)" />
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:39]: What is the value of 0 doing? Doc in help?

File: tools/merqury/merqury_plot_CN.xml [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+.. class:: infomark
+
+**Purpose**
+Make spectra-cn plots. Line, filled, and stacked spectra-cn plots will be
+generated.
+ 
+ --- 
+
+usage: /galaxy/home/delphinel/conda/envs/merqury/share/merqury/plot/plot_spectra_cn.R
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:104]: Remove this path.

File: tools/merqury/merqury_plot_CN.xml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+       [-h] [-f FILE] [-o OUTPUT] [-z ZERO_HIST] [-l CUTOFF] [-x XDIM]
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:105]: All this syntax confuses user. Maybe just move to the parameters help attributes.

File: tools/merqury/merqury_plot_CN.xml [hunk 7]
```diff
@@ -33,18 +33,18 @@
...
     <inputs>
-        <param type="data" name="input1" format="tab" />
-        <param name="xdim" type="integer" label="Width of plot"  value="6" min="0" help="(-x)" />
-        <param name="ydim" type="integer" label="Height of plot"  value="5" min="0"  help="(-y)" />
-        <param name="xmax" type="float" label="Maximum limit for k-mer multiplicity (leave 0 for default)"  value="0" min="0" help="(-m)" />
-        <param name="ymax" type="float" label="Maximum limit for k-mer count (leave 0 for default)"  value="0" min="0"  help="(-n)" />
-        <param name="asm_only" type="data" label="Assembly only counts" format="tab" help="(-z)" optional="true" />
-        <param name="cutoffs" type="data" label="Solid k-mer cutoffs" format="txt" help="(-l)" optional="true" />
+        <param type="data" name="input1" format="tabular" />
+        <param name="xdim" type="integer" label="Width of plot"  value="6" min="0" />
```
Comment (bernt-matthias) [inline tools/merqury/merqury_plot_CN.xml:37]: Can you do this for all parameters:

```suggestion
        <param argument="-f" type="data" name="input1" format="tabular" />
        <param argument="-x" name="xdim" type="integer" label="Width of plot"  value="6" min="0" />
```

File: tools/merqury/merqury_plot_CN.xml [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="bio_tools"/>
+    <expand macro="requirements"/>
+    <version_command>merqury --version</version_command>
+
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir -p output_files/ &&
+        \$MERQURY/plot/plot_spectra_cn.R
```
Comment (bgruening) [inline tools/merqury/merqury_plot_CN.xml:12]: @Delphine-L where is this ` \$MERQURY` coming from?

Comment (bgruening) [reply tools/merqury/merqury_plot_CN.xml:12]: And this is what I do not understand. The tests should run in docker and this variable should not be set. :thinking:

Comment (bgruening) [reply tools/merqury/merqury_plot_CN.xml:12]: Ok, have it. It's an activate script. I'm not really a fan of this, but I think its ok. Thanks @Delphine-L

**Review by bgruening — APPROVED** [review]

**Pull request merged**